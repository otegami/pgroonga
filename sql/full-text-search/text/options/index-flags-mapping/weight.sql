CREATE TABLE memos (
  title text,
  content text
);

CREATE INDEX pgrn_index ON memos
  USING pgroonga (title, content)
  WITH (index_flags_mapping = '{
	  "content": ["WITH_WEIGHT", "WEIGHT_FLOAT32"]
	}');

\out null
SELECT
  'Lexicon' || 'pgrn_index'::regclass::oid || '_0' as title_lexicon_name,
  'Lexicon' || 'pgrn_index'::regclass::oid || '_1' as content_lexicon_name;
\out
\gset

SELECT pgroonga_command('schema')::jsonb#>'{1,tables}'
         ->:'title_lexicon_name'
         #>'{columns,index,command,arguments,flags}';
SELECT pgroonga_command('schema')::jsonb#>'{1,tables}'
         ->:'content_lexicon_name'
         #>'{columns,index,command,arguments,flags}';

DROP TABLE memos;
