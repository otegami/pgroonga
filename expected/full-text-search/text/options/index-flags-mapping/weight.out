CREATE TABLE memos (
  title text,
  content text
);
CREATE INDEX pgrn_index ON memos
  USING pgroonga (title, content)
  WITH (index_flags_mapping = '{
	  "content": ["WITH_WEIGHT", "WEIGHT_FLOAT32"]
	}');
\out null
SELECT
  'Lexicon' || 'pgrn_index'::regclass::oid || '_0' as title_lexicon_name,
  'Lexicon' || 'pgrn_index'::regclass::oid || '_1' as content_lexicon_name;
\out
\gset
SELECT pgroonga_command('schema')::jsonb#>'{1,tables}'
         ->:'title_lexicon_name'
         #>'{columns,index,command,arguments,flags}';
           ?column?           
------------------------------
 "COLUMN_INDEX|WITH_POSITION"
(1 row)

SELECT pgroonga_command('schema')::jsonb#>'{1,tables}'
         ->:'content_lexicon_name'
         #>'{columns,index,command,arguments,flags}';
                        ?column?                         
---------------------------------------------------------
 "COLUMN_INDEX|WITH_WEIGHT|WITH_POSITION|WEIGHT_FLOAT32"
(1 row)

DROP TABLE memos;
